pytorch_lightning/utilities/model_summary.py

```diff
@@ -0,0 +1,469 @@
+        '1.2 K'
+        >>> get_human_readable_count(2e6)   # (two million)
+        '2.0 M'
+        >>> get_human_readable_count(3e9)   # (three billion)
+        '3.0 B'
+        >>> get_human_readable_count(4e14)  # (four hundred trillion)
+        '400 T'
+        >>> get_human_readable_count(5e15)  # (more than trillion)
+        '5,000 T'
+
+    Args:
+        number: a positive integer number
+
+    Return:
+        A string formatted according to the pattern described above.
+
+    """
+    assert number >= 0
+    labels = PARAMETER_NUM_UNITS
+    num_digits = int(np.floor(np.log10(number)) + 1 if number > 0 else 1)
+    num_groups = int(np.ceil(num_digits / 3))
+    num_groups = min(num_groups, len(labels))  # don't abbreviate beyond trillions
+    shift = -3 * (num_groups - 1)
+    number = number * (10**shift)
+    index = num_groups - 1
+    if index < 1 or number >= 100:
+        return f"{int(number):,d} {labels[index]}"
+
+    return f"{number:,.1f} {labels[index]}"
+
+
+def _is_lazy_weight_tensor(p: Tensor) -> bool:
+    if _TORCH_GREATER_EQUAL_1_8:
+        from torch.nn.parameter import UninitializedParameter
+        if isinstance(p, UninitializedParameter):
+            warning_cache.warn(
+                "A layer with UninitializedParameter was found. "
+                "Thus, the total number of parameters detected may be inaccurate."
+            )
+            return True
+    return False
+
+
+def summarize(lightning_module, mode: Optional[str] = "top", max_depth: Optional[int] = None) -> Optional[ModelSummary]:
+    """
+    Summarize the LightningModule specified by `lightning_module`.
+
+    Args:
+        lightning_module: `LightningModule` to summarize.
+        mode: Can be either ``'top'`` (summarize only direct submodules) or ``'full'`` (summarize all layers).
+
+            .. deprecated:: v1.4
+                This parameter was deprecated in v1.4 in favor of `max_depth` and will be removed in v1.6.
+
+        max_depth: The maximum depth of layer nesting that the summary will include. A value of 0 turns the
+            layer summary off. Default: 1.
+
+    Return:
+        The model summary object
+    """
+    model_summary = None
+
+    # temporary mapping from mode to max_depth
+    if max_depth is None:
+        if mode in ModelSummary.MODES:
+            max_depth = ModelSummary.MODES[mode]
+            rank_zero_deprecation(
+                f"Argument `mode` in `LightningModule.summarize` is deprecated in v1.4"
+                f" and will be removed in v1.6. Use `max_depth={max_depth}` to replicate `mode={mode}` behavior."
+            )
+            model_summary = ModelSummary(lightning_module, max_depth=max_depth)
+        elif mode is not None:
+            raise MisconfigurationException(f"`mode` can be None, {', '.join(ModelSummary.MODES)}, got {mode}")
+    else:
+        model_summary = ModelSummary(lightning_module, max_depth=max_depth)
+
+    log.info("\n" + str(model_summary))
```

Comment:
we probably should leave this to the caller to print to avoid this having other side-effects. that way the caller can cleanly control what destination/output they want to send this to (e.g. stdout, file, etc)